internal/ingress/store/store.go

```diff
@@ -160,11 +164,30 @@ func (s Store) GetService(namespace, name string) (*apiv1.Service, error) {
 	return service.(*apiv1.Service), nil
 }
 
-// ListIngresses returns the list of Ingresses
-func (s Store) ListIngresses() []*networkingv1beta1.Ingress {
+// ListIngressesV1 returns the list of Ingresses in the Ingress v1 store.
+func (s Store) ListIngressesV1() []*networkingv1.Ingress {
+	// filter ingress rules
+	var ingresses []*networkingv1.Ingress
+	for _, item := range s.stores.IngressV1beta1.List() {
+		ing, ok := item.(*networkingv1.Ingress)
+		if !ok {
+			s.logger.Warnf("listIngressesV1: dropping object of unexpected type: %#v", item)
+			continue
+		}
+		if !s.isValidIngressClass(&ing.ObjectMeta, s.ingressClassMatching) { // TODO
```

Comment:
Can you add in a card and add it to the 0.10 milestone to track this TODO. It is very very important that we get this right in the first go.

Reply:
There is one already: #790 .

I added a comment there referencing this TODO.

Also, improved the TODO message.